zerver/tests/test_upload.py

```diff
@@ -1203,11 +1203,25 @@ def test_get_gravatar_avatar(self) -> None:
             redirect_url = response["Location"]
             self.assertEqual(redirect_url, str(avatar_url(cordelia)) + "&foo=bar")
 
+        with self.settings(
+            ENABLE_GRAVATAR=False, GRAVATAR_REALM_OVERRIDE={get_realm("zulip").id: True}
+        ):
+            response = self.client_get("/avatar/cordelia@zulip.com", {"foo": "bar"})
+            redirect_url = response["Location"]
+            self.assertEqual(redirect_url, str(avatar_url(cordelia)) + "&foo=bar")
+
         with self.settings(ENABLE_GRAVATAR=False):
             response = self.client_get("/avatar/cordelia@zulip.com", {"foo": "bar"})
             redirect_url = response["Location"]
             self.assertTrue(redirect_url.endswith(str(avatar_url(cordelia)) + "&foo=bar"))
 
+        with self.settings(
+            ENABLE_GRAVATAR=True, GRAVATAR_REALM_OVERRIDE={get_realm("zulip").id: False}
+        ):
+            response = self.client_get("/avatar/cordelia@zulip.com", {"foo": "bar"})
+            redirect_url = response["Location"]
+            self.assertTrue(redirect_url.endswith(str(avatar_url(cordelia)) + "&foo=bar"))
```

Comment:
I don't see how these tests verify that the `GRAVATAR_REALM_OVERRIDE` was applied properly; it seems like the comparisons in the True/False cases are too similar?

Reply:
I copied the test case for `ENABLE_GRAVATAR=False` and checked if the same test passes if I overwrite it with `GRAVATAR_REALM_OVERRIDE` and vice versa.

Reply:
So you want me to more explicitly check if a gravatar URL is returned or not?

Reply:
Yeah, the `endswith` check just seems kinda weak.

Reply:
@timabbott this should be fine now, I am now checking if the url contains `"gravatar"`